[ALM Accelerator - BUG]: export-solution-to-git pipeline fails at step Update Deployment Settings

### Describe the issue

I have set up our ALM Accelerator with validation, test and production environments. All three environments specified by their Environment URL (i.e. `https://<urlname>.crm<region>.dynamics.com/`).

I am exporting a solution from a developer environment. I have set up ALM accelerator to be able to talk to this dev-environment as well (running `coe alm connection add ...`).

I have looked for the latest release on the repository https://github.com/microsoft/coe-alm-accelerator-templates and found https://github.com/microsoft/coe-alm-accelerator-templates/releases/tag/CenterofExcellenceALMAccelerator-August2022-HF1 (at the time of writing 12 days old). I pushed `tags/CenterofExcellenceALMAccelerator-August2022-HF1` into `main` in our Azure DevOps azure-pipelines repository, so that ALM Accelerator uses the templates from that release.

When pressing the `Commit Solution` button in the `ALM Accelerator for PowerPlatform` app the pipeline `export-solution-to-git` is launched and completes all steps successfully until reaching the step `Update Deployment Settings` where it fails. From what I can see in the output it fails immediately after `tsc` has built `coe@0.0.1` (presumably when executing the newly built `coe` tool).

Output (actual environment urls redacted):

```
> coe@0.0.1 build
> tsc


added 1 package, and audited 3 packages in 520ms

found 0 vulnerabilities
Environments:  validation=https://<...>-validation.crm<...>.dynamics.com/,test=https://<...>-test.crm<...>.dynamics.com/,prod=https://<...>.crm<...>.dynamics.com/
info: Branch start
info: Setup branch
info: Using supplied access token
info: Pipeline Project: undefined
info: An error occurred while creating the branch: Error: ﻿{"$id":"1","innerException":null,"message":"A potentially dangerous Request.Path value was detected from the client (:).","typeName":"System.Web.HttpException, System.Web","typeKey":"HttpException","errorCode":0,"eventId":0}
error: ﻿{"$id":"1","innerException":null,"message":"A potentially dangerous Request.Path value was detected from the client (:).","typeName":"System.Web.HttpException, System.Web","typeKey":"HttpException","errorCode":0,"eventId":0} {"statusCode":400}
Checking for secrets removed
Secrets removed creating deploymentConfiguration for Validation


    Directory: D:\a\1\s\alm-accelerator-test\TestSolution\config


Mode                 LastWriteTime         Length Name                                                                 
----                 -------------         ------ ----                                                                 
d-----         8/29/2022   6:58 AM                Validation                                                           
Checking for secrets removed
Secrets removed creating deploymentConfiguration for Test
d-----         8/29/2022   6:58 AM                Test                                                                 
Checking for secrets removed
Secrets removed creating deploymentConfiguration for Prod
d-----         8/29/2022   6:58 AM                Prod                                                                 


##[error]PowerShell exited with code '1'.
```

### Expected Behavior

The pipeline `export-solution-to-git` should complete successfully to enable the subsequent steps in the ALM workflow.

### What component are you experiencing the issue with?

ALM Accelerator Pipelines

### What solution version are you using?

1.0.20220726.1

### Steps To Reproduce

1. Set up CoE environment, install CoE starter kit and ALM Accelerator
1. Set up validation, test and prod environments
1. run `coe alm generate install -o install.json`, supply values for the validation, test, and prod environments, supply CoE for the managed solution environment. Install `all` components.
1. run `coe alm install -f install.json`
1. Set up development environment. Add the ALM accelerator service principal as an S2S App user, create a solution with e.g. one cloud flow in it.
1. run `coe alm connection add` to create a connection to the dev environment
1. play the ALM Accelerator App, select the dev environment, set up a deployment profile for your solution.
1. Click on the `Commit solution` button next to your solution.

### Anything else?

_No response_

--- a/coe-cli/src/common/environment.ts
+++ b/coe-cli/src/common/environment.ts
@@ -157,7 +157,7 @@ export abstract class Environment {
             if ( urlOrg.hostname.toLowerCase() == "dev.azure.com") {
                 orgName = urlOrg.pathname.split('/')[1]
             }
-            else if (legacyUrlOrgMatch = urlOrg.hostname.match(/^(?<orgName>\w+)\.visualstudio\.com$/iu)) {
+            else if (legacyUrlOrgMatch = urlOrg.hostname.match(/^(?<orgName>[^\.]+)\.visualstudio\.com$/iu)) {
                 ({ orgName } = legacyUrlOrgMatch.groups);
             }
         } catch {

--- a/coe-cli/src/common/environment.ts
+++ b/coe-cli/src/common/environment.ts
@@ -152,9 +152,13 @@ export abstract class Environment {
         let orgName = typeof args === "string" ? args : args.organizationName
         try {
             let urlOrg = new url.URL(orgName)
+            let legacyUrlOrgMatch;
             
             if ( urlOrg.hostname.toLowerCase() == "dev.azure.com") {
                 orgName = urlOrg.pathname.split('/')[1]
+            }
+            else if (legacyUrlOrgMatch = urlOrg.hostname.match(/^(?<orgName>\w+)\.visualstudio\.com$/iu)) {
+                ({ orgName } = legacyUrlOrgMatch.groups);
             }
         } catch {
 

